### PR TITLE
Upgrade bumpVersion Github Action to manage native versions too.

### PR DESCRIPTION
**Pull request by roryabraham**
All the context for this PR is in [this slack thread](https://expensify.slack.com/archives/C03V9A4TB/p1612984615476000).

### Details
1) Remove the `postversion` npm script to run `react-native-version`
2) Run `react-native-version` from within the `bumpVersion` script (the only place we should be altering the npm version
3) Use this versioning schema for android: `1.0.1-444` ---> `001000001444` aka 12 digits, three each for `MAJOR`, `MINOR`, `PATCH`, `BUILD`.
4) For iOS, just use the semver version: `1.0.1-444` ---> `1.0.1-444`.

### Fixed Issues
https://github.com/Expensify/Expensify/issues/155197

### Tests
We can test by:

1) Pushing a tag for the current highest version + 1
2) Merging a PR

That should cause the "race condition" to occur. From there, we just need to make sure that the native versions are bumped +2 to match the new npm version

### Tested On

- Github

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
     extends: 'expensify',
     parser: 'babel-eslint',
-    ignorePatterns: ['src/vendor', '!.github/actions/*.js'],
+    ignorePatterns: ['src/vendor', '!.github/actions/**/*.js'],
```
Comment (roryabraham) [inline .eslintrc.js:4]: This still isn't linting the js files in `.github/actions/` 😓

**Comment by roryabraham** [comment]
Also, Verify Github Action Builds keeps failing. I can't parse through the diff very easily and I'm not sure why there even is a diff. Maybe it has to do with the `semver` dependency I added being a dev dependency? I'm really not too sure. I might try to use diff_highlight to get  character-by-character diff highlighting like Github has, and maybe that will get me closer to figuring this out.

File: .eslintrc.js (hunk 1 above)
Comment (AndrewGable) [reply .eslintrc.js:4]: Sounds like a blocker for the external QA actions, right?

Comment (roryabraham) [reply .eslintrc.js:4]: Potentially. Probably not a blocker per-se, but definitely a nice-to-have.

**Comment by roryabraham** [comment]
Figured out why this isn't working ... for some reason `ncc` is including a local path to `Expensify.cash` in the compiled script (i.e: `/Users/roryabraham/ReactNativeChat`), which we obviously don't want.

**Comment by roryabraham** [comment]
Also discovered that it's the `react-native-version` import that causes `/Users/roryabraham/ReactNativeChat` to be included in the source code.

**Comment by roryabraham** [comment]
I think [this issue](https://github.com/vercel/webpack-asset-relocator-loader/issues/112) is the root of the problem. `react-native-version` is internally using `path.cwd()`, and `ncc` is treating that as a static value.

**Comment by roryabraham** [comment]
Also, I was only able to figure this out after copy/pasting the output diff into a more useful diff generator that creates a character-by-character diff. So improving the diff output by `verifyGithubActionBuilds.sh` will be a big improvement.

File: .eslintrc.js (hunk 1 above)
Comment (AndrewGable) [reply .eslintrc.js:4]:
```suggestion
    ignorePatterns: ['!.*', 'src/vendor', '.github/actions/**/index.js'],
```

File: .github/actions/bumpVersion/bumpVersion.js [hunk 2]
```diff
@@ -2,33 +2,120 @@ const {exec} = require('child_process');
...
+    updateNativeVersion('ios', cleanNewVersion);
+}
+
 const MAX_RETRIES = 10;
 let errCount = 0;
 let shouldRetry;
 
 do {
     shouldRetry = false;
+    // eslint-disable-next-line no-loop-func
```
Comment (roryabraham) [inline .github/actions/bumpVersion/bumpVersion.js:98]: Do we feel comfortable suppressing this lint error?

Comment (AndrewGable) [reply .github/actions/bumpVersion/bumpVersion.js:98]: Is there any way around it? It seems like it's necessary, right?

Comment (roryabraham) [reply .github/actions/bumpVersion/bumpVersion.js]: Yeah, I think it's necessary in this case.

**Comment by roryabraham** [comment]
Okay, this is updated and ready for re-review.

**Review by AndrewGable — APPROVED** [review]

**Pull request merged**